app/src/main/java/org/dosomething/letsdothis/ui/fragments/CategoryFragment.java

```diff
@@ -133,13 +143,34 @@ public void onCampaignExpanded(int position)
         recyclerView.smoothScrollToPosition(position);
     }
 
+    @Override
+    public void onReportBackClicked(int reportBackId)
+    {
+        startActivity(ReportBackDetailsActivity.getLaunchIntent(getActivity(), reportBackId));
+    }
+
+    @Override
+    public void onScrolledToBottom()
+    {
+        if(currentPage < totalPages)
+        {
+            Toast.makeText(getActivity(), "get more data", Toast.LENGTH_SHORT).show();
+            ReportBackListTask task = new ReportBackListTask(position,
+                                                             StringUtils.join(sampleIdsSubset, ","),
+                                                             currentPage + 1);
+            TaskQueue.loadQueueDefault(getActivity()).execute(task);
+        }
+    }
+
     @SuppressWarnings("UnusedDeclaration")
     public void onEventMainThread(ReportBackListTask task)
     {
         if(task.position == position)
         {
+            totalPages = task.totalPages;
+            currentPage = task.page;
             List<ReportBack> reportBacks = task.reportBacks;
-            if(reportBacks != null && ! reportBacks.isEmpty())
+            if(reportBacks != null)
```

Comment:
is this ever going to be null? if not then add a FIXME to add a crashlytics. Dont know if we would show the user the fail message